FAILED pandas/tests/indexes/test_common.py::test_sort_values_invalid_na_position[string-pyarrow-None] - AssertionError: Did not see expected warning of class 'PerformanceWarning'
FAILED pandas/tests/indexes/test_common.py::test_sort_values_invalid_na_position[string-pyarrow-middle] - AssertionError: Did not see expected warning of class 'PerformanceWarning'

--- a/pandas/tests/extension/test_arrow.py
+++ b/pandas/tests/extension/test_arrow.py
@@ -36,6 +36,7 @@ from pandas.compat import (
     pa_version_under9p0,
     pa_version_under11p0,
 )
+from pandas.errors import PerformanceWarning
 
 from pandas.core.dtypes.common import is_any_int_dtype
 
@@ -678,7 +679,23 @@ class TestBaseInterface(base.BaseInterfaceTests):
 
 
 class TestBaseMissing(base.BaseMissingTests):
-    pass
+    def test_fillna_no_op_returns_copy(self, data):
+        data = data[~data.isna()]
+
+        valid = data[0]
+        result = data.fillna(valid)
+        assert result is not data
+        self.assert_extension_array_equal(result, data)
+        with tm.assert_produces_warning(PerformanceWarning):
+            result = data.fillna(method="backfill")
+        assert result is not data
+        self.assert_extension_array_equal(result, data)
+
+    def test_fillna_series_method(self, data_missing, fillna_method):
+        with tm.maybe_produces_warning(
+            PerformanceWarning, fillna_method is not None, check_stacklevel=False
+        ):
+            super().test_fillna_series_method(data_missing, fillna_method)
 
 
 class TestBasePrinting(base.BasePrintingTests):

--- a/pandas/tests/extension/test_string.py
+++ b/pandas/tests/extension/test_string.py
@@ -159,17 +159,24 @@ class TestMissing(base.BaseMissingTests):
         self.assert_extension_array_equal(result, expected)
 
     def test_fillna_no_op_returns_copy(self, data):
+        data = data[~data.isna()]
+
+        valid = data[0]
+        result = data.fillna(valid)
+        assert result is not data
+        self.assert_extension_array_equal(result, data)
+
         with tm.maybe_produces_warning(
-            PerformanceWarning,
-            pa_version_under7p0 and data.dtype.storage == "pyarrow",
-            check_stacklevel=False,
+            PerformanceWarning, data.dtype.storage == "pyarrow"
         ):
-            super().test_fillna_no_op_returns_copy(data)
+            result = data.fillna(method="backfill")
+        assert result is not data
+        self.assert_extension_array_equal(result, data)
 
     def test_fillna_series_method(self, data_missing, fillna_method):
         with tm.maybe_produces_warning(
             PerformanceWarning,
-            pa_version_under7p0 and data_missing.dtype.storage == "pyarrow",
+            fillna_method is not None and data_missing.dtype.storage == "pyarrow",
             check_stacklevel=False,
         ):
             super().test_fillna_series_method(data_missing, fillna_method)

--- a/pandas/tests/indexes/test_common.py
+++ b/pandas/tests/indexes/test_common.py
@@ -435,15 +435,8 @@ class TestCommon:
 
 @pytest.mark.parametrize("na_position", [None, "middle"])
 def test_sort_values_invalid_na_position(index_with_missing, na_position):
-    dtype = index_with_missing.dtype
-    warning = (
-        PerformanceWarning
-        if dtype.name == "string" and dtype.storage == "pyarrow"
-        else None
-    )
     with pytest.raises(ValueError, match=f"invalid na_position: {na_position}"):
-        with tm.assert_produces_warning(warning):
-            index_with_missing.sort_values(na_position=na_position)
+        index_with_missing.sort_values(na_position=na_position)
 
 
 @pytest.mark.parametrize("na_position", ["first", "last"])

--- a/pandas/tests/indexes/test_common.py
+++ b/pandas/tests/indexes/test_common.py
@@ -13,7 +13,6 @@ import numpy as np
 import pytest
 
 from pandas.compat import IS64
-from pandas.errors import PerformanceWarning
 
 from pandas.core.dtypes.common import (
     is_integer_dtype,
@@ -437,12 +436,7 @@ class TestCommon:
 @pytest.mark.parametrize("na_position", [None, "middle"])
 def test_sort_values_invalid_na_position(index_with_missing, na_position):
     with pytest.raises(ValueError, match=f"invalid na_position: {na_position}"):
-        with tm.maybe_produces_warning(
-            PerformanceWarning,
-            getattr(index_with_missing.dtype, "storage", "") == "pyarrow",
-            check_stacklevel=False,
-        ):
-            index_with_missing.sort_values(na_position=na_position)
+        index_with_missing.sort_values(na_position=na_position)
 
 
 @pytest.mark.parametrize("na_position", ["first", "last"])

